docs/using.md

```diff
@@ -466,9 +511,31 @@ spec:
           - --destination=gcr.io/my-project/gohelloworld
 ```
 
-If the TaskSpec is provided, TaskRef is not allowed.
+Input and output resources can also be embedded without creating Pipeline Resources. TaskRun resource can include either Pipeline Resource reference or Pipeline Resource Spec but not both. Below is an example where Git Pipeline Resource Spec is provided as input for TaskRun `read-repo`.
+
+```yaml
+apiVersion: pipeline.knative.dev/v1alpha1
+kind: TaskRun
+metadata:
+  name: read-repo
+spec:
+  taskRef:
+    name: read-task
+  trigger:
+    type: manual
+  inputs:
+    resources:
+      - name: workspace
+        resourceSpec:
+          type: git
+          params:
+            - name: url
+              value: https://github.com/pivotal-nader-ziada/gohelloworld
+```
+
+**Note**: TaskRun can embed both TaskSpec and resource spec at the same time. See [example](../examples/run/task-run-resource-spec.yaml) TaskRun. The `TaskRun` will also serve as a record of the history of the invocations of the `Task`.
```

Comment:
nice, new docs look good! :+1: 

(definitely need to create a task about adding test coverage for the examples in our docs sometime soon :sweat_smile: ...)